miden-lib/asm/sat/kernel.masm

```diff
@@ -396,4 +404,75 @@ export.get_account_vault_commitment
     # prepare the stack for return
     swapw dropw
     # => [COM]
+end
+
+#! Mint an asset from the faucet the transaction is being executed against.
+#!
+#! Panics:
+#! - If the transaction is not being executed against a faucet.
+#! - If the asset being minted is not associated with the faucet the transaction is being executed
+#!   against.
+#! - If the asset is not well formed.
+#! - For fungible faucets if the total issuance after minting is greater than the maximum amount 
+#!   allowed.
+#! - For non-fungible faucets if the non-fungible asset being minted already exists.
+#!
+#! Stack: [ASSET]
+#! Outputs: [ASSET]
+#!
+#! - ASSET is the asset that was minted.
+export.mint_asset
+    # AUTHENTICATION
+    # ---------------------------------------------------------------------------------------------
+    # get the hash of the caller
+    padw caller
+    # => [CALLER, ASSET]
+
+    # make sure the caller is a part of the account interface
+    exec.account::authenticate_procedure
+    # => [CALLER, ASSET]
+
+    # drop the caller
+    dropw
+    # => [ASSET]
```

Comment:
This sequence of instructions seems to be repeated quite a few times in this file. Should we create a helper procedure for this? For example, something like `auth_caller`?

Reply:
I had tried to do this previously but I ran into some complications with authentication.  I'm not sure if there is some issue with the way in which procedures are inlined and the `CALL` instruction?  I will create an issue to implement this and investigate any challenges.  How does that sound?

Reply:
https://github.com/0xPolygonMiden/miden-base/issues/221

Reply:
Ah - interesting! Yes, let's investigate in a separate issue.